be/src/formats/orc/orc_mapping.cpp

```diff
@@ -85,20 +91,30 @@ Status OrcMapping::set_include_column_id_by_type(const OrcMappingPtr& mapping, c
     return Status::OK();
 }
 
-std::unique_ptr<OrcMapping> OrcMappingFactory::build_mapping(const std::vector<SlotDescriptor*>& slot_descs,
-                                                             const orc::Type& root_orc_type, const bool case_sensitve) {
+StatusOr<std::unique_ptr<OrcMapping>> OrcMappingFactory::build_mapping(
+        const std::vector<SlotDescriptor*>& slot_descs, const orc::Type& root_orc_type, const bool case_sensitve,
+        const bool use_orc_column_names, const std::vector<std::string>* hive_column_names) {
     std::unique_ptr<OrcMapping> orc_mapping = std::make_unique<OrcMapping>();
-    Status res = _init_orc_mapping(orc_mapping, slot_descs, root_orc_type, case_sensitve);
+    Status res;
+    if (use_orc_column_names) {
+        // Only used for broker load and UT test now
+        res = _init_orc_mapping(orc_mapping, slot_descs, root_orc_type, case_sensitve, use_orc_column_names);
+    } else {
+        res = _init_orc_mapping(orc_mapping, slot_descs, root_orc_type, case_sensitve, use_orc_column_names,
+                                hive_column_names);
+    }
     if (!res.ok()) {
-        LOG(WARNING) << res.to_string();
-        return nullptr;
+        return res;
     }
     return orc_mapping;
 }
 
 Status OrcMappingFactory::_init_orc_mapping(std::unique_ptr<OrcMapping>& mapping,
                                             const std::vector<SlotDescriptor*>& slot_descs,
-                                            const orc::Type& orc_root_type, const bool case_sensitve) {
+                                            const orc::Type& orc_root_type, const bool case_sensitve,
```

Comment:
Why two different functions? I think it could be a single function with if-else inside.

Reply:
Two function's mapping method is different a lot, I think if put into one function, the code may be complex for people to understand.

Reply:
Then I suggest to use different names to avoid confusion. Or to add some comments to explicitly say what's for.

Reply:
Good idea